crates/matrix-sdk/src/notification_settings/mod.rs

```diff
@@ -323,14 +323,45 @@ mod tests {
         client: &Client,
         rules: Vec<(RuleKind, &RoomId, bool)>,
     ) -> NotificationSettings {
-        let ruleset = get_server_default_ruleset();
-        // XXX would be slightly better to only use `Ruleset` here too, and no
-        // `RuleCommands` so we're testing things more in isolation.
-        let mut rule_commands = RuleCommands::new(ruleset);
+        let mut ruleset = get_server_default_ruleset();
         for (kind, room_id, notify) in rules {
-            rule_commands.insert_rule(kind, room_id, notify).unwrap();
+            let actions = if notify {
+                vec![Action::Notify, Action::SetTweak(Tweak::Sound("default".into()))]
+            } else {
+                vec![]
+            };
+            match kind {
+                RuleKind::Room => {
+                    let new_rule = NewSimplePushRule::new(room_id.to_owned(), actions);
+                    ruleset.insert(NewPushRule::Room(new_rule), None, None).unwrap();
+                }
+                RuleKind::Override => {
+                    let new_rule = NewConditionalPushRule::new(
+                        room_id.into(),
+                        vec![PushCondition::EventMatch {
+                            key: "room_id".to_owned(),
+                            pattern: room_id.to_string(),
+                        }],
+                        actions,
+                    );
+                    ruleset.insert(NewPushRule::Override(new_rule), None, None).unwrap();
+                }
+                RuleKind::Underride => {
+                    let new_rule = NewConditionalPushRule::new(
+                        room_id.into(),
+                        vec![PushCondition::EventMatch {
+                            key: "room_id".to_owned(),
+                            pattern: room_id.to_string(),
+                        }],
+                        actions,
+                    );
+                    ruleset.insert(NewPushRule::Underride(new_rule), None, None).unwrap();
+                }
```

Comment:
These two branches are mostly the same, until the `NewPushRule::Override/Underride` variants that are different; can you put more code in common please, by using a `RuleKind::Override | RuleKind::Underride` pattern, and then defining the `NewPushRule` variant based on the `kind`?

Reply:
Done.